src/pygeom.c

```diff
@@ -9,14 +9,20 @@
 
 
 /* This initializes a pointer to a NULL geometry */
-GeometryObject *Geom_Empty = NULL;
+GeometryObject *NaG = NULL;
 
-/* Initializes a new geometry object, without Empty check */
-PyObject *GeometryObject_FROMGEOS(PyTypeObject *type, GEOSGeometry *ptr)
+/* Initializes a new geometry object */
+PyObject *GeometryObject_FromGEOS(PyTypeObject *type, GEOSGeometry *ptr)
 {
     if (ptr == NULL) {
-        Py_INCREF(Geom_Empty);
-        return (PyObject *) Geom_Empty;
+        if (NaG == NULL) {
+            /* First time: initialize the NaG object */
```

Comment:
Shouldn't we do this first time initialization in the ufuncs module initialization function?